app/controllers/queue_classic_admin/queue_classic_jobs_controller.rb

```diff
@@ -14,20 +15,22 @@ def destroy
     end
 
     def destroy_all
-      filter_jobs(QueueClassicJob).delete_all
+      filter_jobs(QueueClassicJob)
+      @queue_classic_jobs.ready.delete_all
       redirect_to queue_classic_jobs_url
     end
 
     def unlock_all
-      filter_jobs(QueueClassicJob).where('locked_at < ?', 5.minutes.ago).update_all(locked_at: nil)
+      filter_jobs(QueueClassicJob)
+      @queue_classic_jobs.ready.where('locked_at < ?', 5.minutes.ago).update_all(locked_at: nil)
       redirect_to queue_classic_jobs_url
     end
 
     def bulk_custom_action
-      jobs = filter_jobs(QueueClassicJob)
-
+      filter_jobs(QueueClassicJob)
+      @queue_classic_jobs.ready
```

Comment:
Totally agree. Basically, this shouldn't be merged before more work, it's not future proof and super hacky.